src/backend/main.ts

```diff
@@ -160,6 +165,76 @@ export async function bootstrapWorkspace(opts: {
     }
     yDoc.clientID = clientID
   }
+  // Adam Chen - Injection point 4: new member join
+  console.log('-- Injection point 4: new member join check --')
+  const appPrefixName = appPrefix.toString()
+  console.log('debug: appPrefix: ', appPrefixName)
+  console.log('debug: nodeID', nodeId)
+  const snapshotName = appPrefix.append('32=snapshot').toString()
+  console.log('debug: theoretical snapshot prefix: ', snapshotName)
+  const localYJSUpdate = await persistStore.get('localSnapshot')
+  console.log('persistent storage local yjs update: ', localYJSUpdate)
+  // localYJSUpdate = undefined
+
+  // //State Vector Testing
+  // let test = new StateVector()
+  // test.set(new Name('/user1'), 1)
+  // test.set(new Name('/user2'), 1)
+  // let svsComponent = new Component(Version.type, Encoder.encode(test))
+  // console.log('Debug: SVS test: ', svsComponent.toString())
+  // console.log('Debug: parsing SVS', Decoder.decode(svsComponent.value, StateVector))
+
+  if (!localYJSUpdate) {
+    console.log('Empty yjs local storage detected, fetching snapshot')
```

Comment:
Actually you can also fetch if the local state carries an old timestamp.
But good for now.

Reply:
I was just thinking about that. Expanding this condition to cover a state with an older state vector than a discovered snapshot.
This would cut down on the amount of snapshot fetched as an existing member coming online.
There might be a technicality that is if there are outstanding changes that are local (as SVS Data with state vector changed) but not synced.. Which can be worked out 
Let's discuss this further

Reply:
A simplest solution will be just attaching a timestamp. So you do not need to worry about comparing the states.
Though, comparing states may be optimal in practice.

Reply:
Timestamps added as per the snapshot 1.0 discussion.